core/src/com/geekbrains/td/Antiking.java

```diff
@@ -0,0 +1,179 @@
+        this.hitArea.set(position.x, position.y, 32.0f);
+    }
+
+
+    public void getNextPoint() {
+        float destX = destCellX * 80 + 40;
+        float destY = destCellY * 80 + 40;
+        if ((position.x - destX) * (position.x - destX) + (position.y - destY) * (position.y - destY) < 8.0f) {
+            CheckDestPosition();
+        }
+        gameScreen.getMap().buildRoute(destCellX, destCellY, (int) (position.x / 80), (int) (position.y / 80), destination);
+        destination.scl(80, 80).add(40, 40);
+    }
+
+    private void CheckDestPosition() {
+        int code = helperMap[destCellX][destCellY];
+        if (code == 2) {
+            speed = 0.0f;
+        } else {
+            speed = 150.0f;
+            if (!findCell(2)) {
+                for (int i = 0; i < 16; i++) {
+                    destCellX = MathUtils.random(3, 15);
+                    destCellY = MathUtils.random(0, 8);
+                    int newCode = helperMap[destCellX][destCellY];
+                    if (code == 1 && newCode == 0) break;
+                    if (code == 0 && newCode == 1) break;
+                }
+            }
+        }
+    }
+
+    private boolean findCell(int code) {
+        int width = helperMap.length;
+        int height = helperMap[0].length;
+        for (int i = width - 1; i >= 0; i--) {
+            for (int j = height - 1; j >= 0; j--) {
+                if (helperMap[i][j] == code) {
+                    destCellX = i;
+                    destCellY = j;
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+
+    int[][] helperMap;
+
+    public void updateHelperMap() {
+        helperMap = gameScreen.getMap().getMapOfWalls();
+        int width = helperMap.length;
+        int height = helperMap[0].length;
+
+        Rectangle rectangle = new Rectangle();
+        for (Turret t : gameScreen.getTurretEmitter().getActiveList()) {
+            helperMap[t.getCellX()][t.getCellY()] = 5;
+        }
+        for (Turret t : gameScreen.getTurretEmitter().getActiveList()) {
+            float fireRadius2 = t.getFireRadius() * t.getFireRadius();
+            int x = t.getCellX() * 80 + 40;
+            int y = t.getCellY() * 80 + 40;
+            for (int i = 0; i < width; i++) {
+                for (int j = 0; j < height; j++) {
+                    int x1 = i * 80 + 40;
+                    int y1 = j * 80 + 40;
+                    double r2 = (x - x1) * (x - x1) + (y - y1) * (y - y1);
+                    if (r2 < fireRadius2 && helperMap[i][j] != 9) {
+                        if (helperMap[i][j] != 5) helperMap[i][j] = 1;
+                    }
+                }
+            }
+        }
+        for (int i = 0; i < width; i++) {
+            for (int j = 0; j < height; j++) {
+                if (helperMap[i][j] == 1) {
+                    int x1 = i * 80 + 40;
+                    int y1 = j * 80 + 40;
+                    for (int k = 0; k < width; k++) {
```

Comment:
There is a golden rule: if you have more then two nested loops or iterations, your code needs restructuration. 
In this particular case, I would move into a separate method everything starting from `for (int k = 0; k < width; k++)`. Or rethink the algorithm, so you won't have `for(..) { for(..) { for(..) { for(..) {  } } } }`.